### DISCUSSION TITLE
Automatically connect to network via vpn when leaving it

### DISCUSSION THREAD

**Discussion by lx1** [schwabe/ics-openvpn#1473] [Q&A] [answered]
Hi all.

Maybe this might be a FAQ, but I could not find a definitive answer.
I have my home network, and would simply like Openvpn for Android to automatically connect to it when I leave it, and disconnect when I come back home. I'm asking if there is some tested, reliable method to do this.

What I tried so far:
I tried to block access to the VPN gateway when I'm home. But the client did seem to maintain the route to the local network, making it unreachable.
I tried to hide the DNS resolution of the VPN gateway, when I'm home, with similar results. Both with and without keepalives.
I tried Tasker and Easer, two tools for task automation. They both did not work very reliably, in my experience. They did seem to miss some network events. Anyway, since I switched to Android 12 (Evolution X custom rom), it seems that both these tools never receive and never handle the network events any more. Maybe, something changed in A12 about this.

Is there an tested method, to disconnect the vpn when that network is directly connected, and the reverse?

Thank you very much,
Luigi

**Comment by schwabe** [comment, marked as answer]
No there is not. My app does not implement these "VPN only on LTE, only on certain WiFis" since Android currently offers no reliable way to ensure this behaviour. So everything would be delayed by a few seconds or might not work at all, leaving the network without VPN for that time.

**Reply by Humara999** [reply]
Hi my vip

**Reply by Humara999** [reply]
#1524

**Comment by lx1** [comment]
Thank you Author. At the very least, this marks a certain answer, even if it is not what I hoped for😃 Thank you as always for the software.
